apphub/lr_controller/super_convergence/super_convergence_tf.py

```diff
@@ -0,0 +1,197 @@
+# Copyright 2019 The FastEstimator Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+import tempfile
+
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+import fastestimator as fe
+from fastestimator.backend import get_lr
+from fastestimator.dataset.data.cifair10 import load_data
+from fastestimator.op.numpyop.meta import Sometimes
+from fastestimator.op.numpyop.multivariate import HorizontalFlip, PadIfNeeded, RandomCrop
+from fastestimator.op.numpyop.univariate import CoarseDropout, Normalize, Onehot
+from fastestimator.op.tensorop.loss import CrossEntropy
+from fastestimator.op.tensorop.model import ModelOp, UpdateOp
+from fastestimator.trace import Trace
+from fastestimator.trace.adapt import LRScheduler
+from fastestimator.trace.io import BestModelSaver
+from fastestimator.trace.metric import Accuracy
+from fastestimator.util.util import Suppressor
+
+
+def super_schedule(step, lr_max, lr_min, mid, end):
+    if step < mid:
+        lr = step / mid * (lr_max - lr_min) + lr_min  # linear increase from lr_min to lr_max
+
+    elif mid <= step < mid * 2:
+        lr = lr_max - (step - mid) / mid * (lr_max - lr_min)  # linear decrease from lr_max to lr_min
+
+    else:
+        lr = max(lr_min - (step - 2 * mid) / (end - 2 * mid) * lr_min, 0)  # linear decrease from lr_min to 0
+
+    return lr
+
+
+def linear_increase(step, min_lr=0.0, max_lr=3.0, num_steps=5000):
+    lr = step / num_steps * (max_lr - min_lr) + min_lr
+    return lr
+
+
+def residual(x, num_channel):
+    x = layers.Conv2D(num_channel, 3, padding='same')(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    x = layers.Conv2D(num_channel, 3, padding='same')(x)
+    x = layers.BatchNormalization(momentum=0.8)(x)
+    x = layers.LeakyReLU(alpha=0.1)(x)
+    return x
+
+
+def my_model():
+    # prep layers
```

Comment:
if we're going to keep re-using this model, should we add it to the FE architecture directory?

Reply:
Maybe? Is this a well-known model?

Reply:
I created a task in project board.